File: OneSignalSDK/onesignal/src/main/java/com/onesignal/language/LanguageProviderAppDefined.java
package com.onesignal.language;
import android.support.annotation.NonNull;

import com.onesignal.OSSharedPreferences;

public class LanguageProviderAppDefined implements LanguageProvider {
    public static final String PREFS_OS_LANGUAGE = "PREFS_OS_LANGUAGE";

    private static final String DEFAULT_LANGUAGE = "en";
    private final OSSharedPreferences preferences;

    public LanguageProviderAppDefined(OSSharedPreferences preferences) {
        this.preferences = preferences;
    }

    public void setLanguage(String language) {
        preferences.saveString(
                preferences.getPreferencesName(),
                PREFS_OS_LANGUAGE,
                language);
    }

    @NonNull
    public String getLanguage() {
        return preferences.getString(
                preferences.getPreferencesName(), PREFS_OS_LANGUAGE, DEFAULT_LANGUAGE);
    }
}

File: OneSignalSDK/onesignal/src/main/java/com/onesignal/onesignal/core/internal/language/impl/LanguageProviderAppDefined.kt
package com.onesignal.onesignal.core.internal.language.impl

import com.onesignal.onesignal.core.internal.language.ILanguageProvider
import com.onesignal.onesignal.core.internal.preferences.IPreferencesService
import com.onesignal.onesignal.core.internal.preferences.PreferenceStores

class LanguageProviderAppDefined(
    private val preferences: IPreferencesService
) : ILanguageProvider {

    override var language: String
        get() = preferences.getString(PreferenceStores.ONESIGNAL, PREFS_OS_LANGUAGE, DEFAULT_LANGUAGE)!!
        set(language) {
            preferences.saveString(PreferenceStores.ONESIGNAL, PREFS_OS_LANGUAGE, language)
        }

    companion object {
        const val PREFS_OS_LANGUAGE = "PREFS_OS_LANGUAGE"
        private const val DEFAULT_LANGUAGE = "en"
    }
}